app/controllers/admin/collections_controller.rb

```diff
@@ -326,7 +326,11 @@ def build_default_lending_period(collection)
       collection.errors.add(:lending_period, "hours needs to be a positive integer.")
     end
 
-    flash[:notice] = collection.errors.full_messages.join if collection.errors.present?
+    if lending_period.zero?
+      collection.errors.add(:lending_period, "cannot be set to 0 days 0 hours.")
+    end
+
+    flash[:notice] = collection.errors.full_messages.join if collection.errors.present? && Settings.controlled_digital_lending.enable
```

Comment:
I think it would make more sense to add the conditional around the building/setting of the default lending period on line 302-303.